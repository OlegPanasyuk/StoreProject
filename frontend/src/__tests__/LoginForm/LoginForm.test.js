import React from 'react';
import LoginFormConnected, {LoginForm} from '../../LoginForm/LoginForm';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
// import rest from 'rest';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

// import pathPrefix from 'rest/interceptor/pathPrefix';
// import errorCode from 'rest/interceptor/errorCode';
// import mime from 'rest/interceptor/mime';

// const client = rest.wrap(mime, { mime: 'application/json' })
//     .wrap(errorCode, { code: 500 })
//     .wrap(pathPrefix, { prefix: 'http://localhost:3300' });


function setupComponent(props = {
    handleConverStatusUser: ()=>{},
    handleSetStateInApp: ()=>{},
    userState: {
        email: 'test@test.com',
        role: 'Guest'
    }
}) {
    let component = shallow(<LoginForm {...props} />);
    return { component, props };
}


describe('LoginForm', () => {

    it('Component must be created', () => {
        const { component } = setupComponent();

        expect(LoginForm.propTypes.userState).toBe(PropTypes.object);

        expect(LoginForm.propTypes.handleConverStatusUser).toBe(PropTypes.func);

        expect(LoginForm.propTypes.handleSetStateInApp).toBe(PropTypes.func);

        expect(shallowToJson(component)).toMatchSnapshot();

    });
});