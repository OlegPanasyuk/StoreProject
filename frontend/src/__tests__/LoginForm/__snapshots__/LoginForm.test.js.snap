// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component must be created 1`] = `
<Form
  as="form"
  className="ml-auto"
  inline={false}
>
  <FormRow
    className="d-flex align-items-center justify-content-end"
  >
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        }
      }
      className="col mb-0"
      controlId="formGridEmail"
    >
      <FormControl
        as="input"
        defaultValue="test@test.com"
        placeholder="Enter email"
        size="sm"
        type="email"
      />
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        }
      }
      className="col mb-0"
      controlId="formGridPassword"
    >
      <FormControl
        as="input"
        placeholder="Password"
        size="sm"
        type="password"
      />
    </FormGroup>
    <Col
      as="div"
      className="d-flex align-items-center"
    >
      <Button
        active={false}
        className=""
        disabled={false}
        onClick={[Function]}
        size="sm"
        type="button"
        variant="secondary"
      >
        Login
      </Button>
      <Link
        ConverStatusUser={[Function]}
        className="d-block p-3"
        href="#"
        text="Registration"
      />
    </Col>
  </FormRow>
</Form>
`;
